plugins/module_utils/_hashi_vault_common.py

```diff
@@ -57,6 +61,20 @@ def get_vault_client(
         :type hashi_vault_revoke_on_logout: bool
         '''
 
+        url = kwargs.get('url') or os.environ.get('VAULT_ADDR')
+        if url and url.startswith('unix://'):
+            try:
+                import requests_unixsocket
+            except ImportError as e:
+                from ansible.module_utils.basic import missing_required_lib
+                raise MissingLibraryError(error=str(e), msg=missing_required_lib('requests_unixsocket'))
+            socket_path = (url
+                           .replace('unix://', '')
+                           .replace('/', '%2F'))
+            socket_url = f"http+unix://{socket_path}"
+            kwargs['url'] = socket_url
+            kwargs['session'] = requests_unixsocket.Session()
```

Comment:
I have some concerns that creating a blank session here is going to end up erasing certain other connection options if they are specified. We have helper methods for this purpose, see:
- https://github.com/ansible-collections/community.hashi_vault/blob/main/plugins/module_utils/_connection_options.py#L98
- https://github.com/ansible-collections/community.hashi_vault/blob/main/plugins/module_utils/_connection_options.py#L123